src/SearchEngine/MASA.BuildingBlocks.SearchEngine.AutoComplete/AudoCompleteDocument.cs

```diff
@@ -0,0 +1,31 @@
+namespace MASA.BuildingBlocks.SearchEngine.AutoComplete;
+public class AudoCompleteDocument<TValue>
+{
+    public string Id { get; set; }
+
+    public string Text { get; set; }
+
+    public TValue Value { get; set; }
+
+
+    public AudoCompleteDocument()
+    {
+    }
+
+    public AudoCompleteDocument(string text, TValue value) : this()
+    {
+        Text = text;
+        Value = value;
+        Id = GetId();
+    }
+
+    public AudoCompleteDocument(string id, string text, TValue value) : this(text, value)
+    {
+        Id = id;
+    }
+
+    protected virtual string GetId()
+    {
+        return $"[{Value}]{Text}";
```

Comment:
I think you can add a IdTemplate property, and default is this. Used by string format.

Reply:
It is not very convenient to use IdTemplate for subsequent extensions, and IdFormatter assignment is not very easy to use. I am going to add:

```
protected virtual string IdGenerator() => $"[{Value}]{Text}";
```

Subsequent expansion is more flexible

Reply:
It is not very convenient to use IdTemplate for subsequent extensions, and IdFormatter assignment is not very easy to use. I am going to add:

```
public class AutoCompleteDocument<TValue>
{
    public string Id => IdGenerator();

    public string Text { get; set; }

    public TValue Value { get; set; }

    public AutoCompleteDocument()
    {
    }

    public AutoCompleteDocument(string text, TValue value) : this()
    {
        Text = text;
        Value = value;
    }

    protected virtual string IdGenerator() => $"[{Value}]{Text}";
}
```